processing/data_collection/gazette/spiders/mg_itauna.py

```diff
@@ -0,0 +1,36 @@
+import datetime as dt
+from dateparser import parse
+from scrapy import Request
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class MgItaunaSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3133808"
+    name = "mg_itauna"
+    allowed_domains = ["itauna.mg.gov.br"]
+    start_urls = ["https://www.itauna.mg.gov.br/portal/diario-oficial/"]
+
+    def parse(self, response):
+        url = self.start_urls[0] + "{i}/0/0/0/0/"
+        last_page = int(response.xpath("//select/option/text()")[-1].extract())
+        for i in range(1, 1 + last_page):
+            yield Request(url.format(i=i), callback=self.parse_editions_page)
+
+    def parse_editions_page(self, response):
```

Comment:
**thought**: The URL you defined in `start_urls` is the equivalent of `https://www.itauna.mg.gov.br/portal/diario-oficial/1/0/0/0/0/` (the first page of the results). Because of that, all the logic that are inside this method could be placed inside `parse` method, and the pagination could start on `2` instead of `1`. The way you did now, add one extra request to the whole scraping process. This is not too bad, but it is better to send as few requests as possible always. It is not a problem to reuse the same callback for processing a page and handling its pagination.

Reply:
@rennerocha and @giuliocc , thanks for the patience and the tips, my idea is to continue with other municipalities here in MG. @rennerocha  in the first version had done it with only one function, but I believe that segmented like that, it was easier to understand.

Reply:
Great, @pgarcias01 . The more hands the better ;)
---
I put this response in another comment made by Renne. Since Github's notifications always confuses me I don't know if you saw it or not so I'll just paste it here :)

> Then maybe adding a `yield from self.parse_editions_page(response)` at the end of the `parse` method and change the range to start at 2?
> 
> This addresses the unnecessary request maintaining separations of concerns.
> 
> I said it could create some confusion because the loop is creating new requests to the same method that creates those same requests. If the dupefilter wasn't in place this would be an infinite loop. And it would be better to address this in an explicit way than a implicit one.
> 
> Should have explained it a bit better.

Reply:
Sorry @rennerocha I got it wrong, now i changed the start of the range